src/test/java/de/bonndan/nivio/observation/FileSourceReferenceObserverTest.java

```diff
@@ -0,0 +1,69 @@
+package de.bonndan.nivio.observation;
+
+import de.bonndan.nivio.input.FileFetcher;
+import de.bonndan.nivio.input.LandscapeDescriptionFactory;
+import de.bonndan.nivio.input.dto.LandscapeDescription;
+import de.bonndan.nivio.input.dto.SourceReference;
+import de.bonndan.nivio.input.http.HttpService;
+import de.bonndan.nivio.util.URLHelper;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import retrofit2.http.Url;
+
+import java.io.File;
+import java.io.IOException;
+import java.net.MalformedURLException;
+import java.net.URL;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.concurrent.ExecutionException;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+class FileSourceReferenceObserverTest {
+
+    private FileFetcher fileFetcher;
+    private LandscapeDescriptionFactory landscapeDescriptionFactory;
+
+    @BeforeEach
+    public void setup() {
+        fileFetcher = new FileFetcher(new HttpService());
+        landscapeDescriptionFactory = new LandscapeDescriptionFactory(fileFetcher);
+    }
+
+    @Test
+    public void canHandleCombinedPath() throws IOException, ExecutionException, InterruptedException {
+
+        String realpath = getRootPath() + "/src/test/resources/example/services/wordpress.yml";
+        File realfile = new File(realpath);
+
+        String source = getRootPath() + "/src/test/resources/example/example_env.yml";
+        File file = new File(source);
+
+        LandscapeDescription description = landscapeDescriptionFactory.fromYaml(file);
+
+        SourceReference reference = description.getSourceReferences().get(0);
+        reference.setUrl("./services/wordpress.yml");
+
+        FileFetcher mocked = mock(FileFetcher.class);
+        when(mocked.get(any(SourceReference.class), any(URL.class))).thenAnswer(invocationOnMock -> String.valueOf(Math.random()));
```

Comment:
Avoiding random numbers would make this test more deterministic.

Doesn't this test fail when the same number is generated 2 times in a row?

Reply:
I've been too lazy to look up this simple code:

`when(mocked.get(any(SourceReference.class), any(URL.class))).thenReturn("foo", "bar")`